### DISCUSSION TITLE
Can you combine different text styles when writing text?

### DISCUSSION THREAD

**Discussion by jez9999** [SixLabors/ImageSharp#2845] [Q&A]
Using the following, I can write a block of text using a font and style of my choosing:

```
img.Mutate(x =>
	x.DrawText(
		new RichTextOptions(_fontManager.GetFont("SomeFont", 100, FontStyle.Regular)) {
			Origin = new Vector2(810, 460),
			LayoutMode = LayoutMode.HorizontalTopBottom,
			HorizontalAlignment = HorizontalAlignment.Center,
			VerticalAlignment = VerticalAlignment.Center,
			TextAlignment = TextAlignment.Center,
			WrappingLength = 800
		},
		"This is a really important point.",
		Color.Black
	)
);
```

However, is it possible for me to write a certain portion of that text in a different style?  If I want to emphasize two words for example:

```
img.Mutate(x =>
	x.DrawText(
		new RichTextOptions(_fontManager.GetFont("SomeFont", 100, FontStyle.Regular)) {
			Origin = new Vector2(810, 460),
			LayoutMode = LayoutMode.HorizontalTopBottom,
			HorizontalAlignment = HorizontalAlignment.Center,
			VerticalAlignment = VerticalAlignment.Center,
			TextAlignment = TextAlignment.Center,
			WrappingLength = 800
		},
		"This is a <b>really important</b> point.",
		Color.Black
	)
);
```

If that's not possible it'd be a nice feature to have.

**Comment by tocsoft** [comment]
Yes, yes you can. What you are looking for is the `TextRuns` property of `RichTextOptions`

From that API it might not be immediate obvious how you would handle specifically bold and italic options, you handle them by setting the `Font` on the text run to a completely separate version of the fond in bold mode. 

What most people do not realise about fonts is that things like bold, italic etc are actually completely separate fonts that applications try and hide away for you.

```c#
var regular = _fontManager.GetFont("SomeFont", 100, FontStyle.Regular);
var bold = _fontManager.GetFont("SomeFont", 100, FontStyle.Bold); // this could be any font not just bold etc
img.Mutate(x =>
	x.DrawText(
		new RichTextOptions(regular) {
			Origin = new Vector2(810, 460),
			LayoutMode = LayoutMode.HorizontalTopBottom,
			HorizontalAlignment = HorizontalAlignment.Center,
			VerticalAlignment = VerticalAlignment.Center,
			TextAlignment = TextAlignment.Center,
			WrappingLength = 800,
                        TextRuns = new[]{
                            new RichTextRun(){ Font = bold, Start = 10, End = 33 }
                        }
		},
		"This is a <b>really important</b> point.",
		Color.Black
	)
);
```

note the inline mark up is doing nothing here it's just the Grapheme index of the `start` and `end` thats important.

**Reply by jez9999** [reply]
Yep, that seems to do what I want (I think you accidentally removed the font from the `RichTextOptions` constructor, by the way), except... I think I discovered a bug in ImageSharp whilst I was working on this.  I've filed it as #2846.

**Reply by tocsoft** [reply]
Fixed my example, yeah when I made the named font variables I forgot to add it back in. Thanks for spotting
